### ISSUE TITLE
Problem about trainsets of ShanghaiTech

### ISSUE THREAD

**Issue by ZiruiYan** [StevenLiuWen/ano_pred_cvpr2018#52]
I found that all videos from ped1, Avenue and the test set of ShanghaiTech of science and technology were extracted into frames. However, Shanghaitech's trainsets are in video format. How to extract it?

**Comment by ZiruiYan** [comment]
Use
ffmpeg -r 1 -i 'input_file' -qmin 1 - qscale:v 1 -r 1 'result_file'
This will give what you want.

**Comment by abhishekaich27** [comment]
@ZiruiYan were you able to make your code work with this preprocessing of Shanghai-Tech training dataset?

**Comment by ZiruiYan** [comment]
Yes, but I didn’t carefully adjust the parameters to achieve high accuracy on that dataset. So I'm not 100% sure that the training set and the test set are extracted with the same parameter.

**Comment by BenedictGreen** [comment]
@ZiruiYan 你好，请问使用ffmpeg把video处理成图像时，参数‘-r’选择多少合适呀，我设置成每秒提取一帧，总共的图片量只有700m，应该不太合适，

**Comment by abhishekaich27** [comment]
@BenedictGreen I have created two scripts that can help you extract these frames according to the author's settings. Let me know if you need them (share your email ID).

**Comment by BenedictGreen** [comment]
@abhishekaich27 Thxs for your reply. I'd appreciate if you send me the scripts.'kuaikuaizhang9700@163.com', that's my e-mail.

**Comment by zzzzyq-18** [comment]
请问您是1s抽取了多少帧呢

**Comment by zzzzyq-18** [comment]
Excuse me for video, are you chose many frames 1 s extraction

**Comment by zzzzyq-18** [comment]
Excuse me, are you how many frames 1 second extraction

**Comment by BenedictGreen** [comment]
12 frames per second. But i'm not sure it is correct.Maybe this processing has produced too many redundant frames

**Comment by Smallwhite999** [comment]
Is there a complete code? On data set preprocessing

**Comment by zdl12345678** [comment]
> 

Can you share a copy with me? Thank you very much, 840771073@qq.com

**Comment by zdl12345678** [comment]
> @BenedictGreen I have created two scripts that can help you extract these frames according to the author's settings. Let me know if you need them (share your email ID).

Can you share a copy with me? Thank you very much, 840771073@qq.com

**Comment by fanzz1208** [comment]
> 

2563992699@qq.com    谢谢

**Comment by hhhh1230kvj** [comment]
> @BenedictGreen I have created two scripts that can help you extract these frames according to the author's settings. Let me know if you need them (share your email ID).

1418319084@qq.com,thanks

**Comment by zzoei** [comment]
> 

739020831@qq.com  thank you！

**Comment by HannoChen** [comment]
> 

Could you share a copy with me, hichen.yh@qq.com, Thank you!

**Comment by HannoChen** [comment]
> @BenedictGreen I have created two scripts that can help you extract these frames according to the author's settings. Let me know if you need them (share your email ID).

Could you share a copy with me, thanks very much! 
email: hichen.yh@qq.com

**Comment by kelummadusanka** [comment]
> @BenedictGreen I have created two scripts that can help you extract these frames according to the author's settings. Let me know if you need them (share your email ID).

@abhishekaich27 can you share a script with me also, thanks 
ikelummadusanka@gmail.com

**Comment by ck-w-0** [comment]
> @BenedictGreen I have created two scripts that can help you extract these frames according to the author's settings. Let me know if you need them (share your email ID).

can you share me alse,thx
wck1916@gmail.com

**Comment by LeoGeralt** [comment]
> @BenedictGreen我创建了两个脚本，可以帮助您根据作者的设置提取这些帧。如果您需要它们，请告诉我（分享您的电子邮件 ID）。

Can you also share it with me?Thank you! 1749953604@qq.com
